### PR TITLE
Ef impementation

### PR DESCRIPTION
**Pull request by AntsiferauGodel**
Implemented EF (task 1-2) no migrations yet

### CODE REVIEW

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/FileStorageContext.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using System.Data.Common;
+using System.Data.Entity;
+using ConsoleFileStorage.Data.Migrations;
+using ConsoleFileStorage.Domain;
+
+namespace ConsoleFileStorage.Data
+{
+    [DbConfigurationType(typeof(SqlConfiguration))]
+    public class FileStorageContext : DbContext
```
Comment (HannaKeryna) [inline Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/FileStorageContext.cs]: What do you think about adding interface for FileStorageContext?

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/Migrations/Configuration.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        public Configuration()
+        {
+            AutomaticMigrationsEnabled = false;
+            ContextKey = "ConsoleFileStorage.Data.FileStorageContext";
+        }
+
+        protected override void Seed(ConsoleFileStorage.Data.FileStorageContext context)
+        {
+            // This method will be called after migrating to the latest version.
```
Comment (HannaKeryna) [inline Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/Migrations/Configuration.cs]: Please remove empty method

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/SqlConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace ConsoleFileStorage.Data
+{
+    public class SqlConfiguration : DbConfiguration
+    {
+        public SqlConfiguration()
+        {
+            SetExecutionStrategy("System.Data.SqlClient", () => new SqlAzureExecutionStrategy());
+            SetDefaultConnectionFactory(new LocalDbConnectionFactory("mssqllocaldb"));
+        }
```
Comment (HannaKeryna) [inline Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/SqlConfiguration.cs]: Please investigate this moment

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/CliRepository.cs [hunk 4]
```diff
@@ -9,39 +8,31 @@ namespace ConsoleFileStorage.Data
...
-        public CliRepository(IDictionary<string, T> dictionary)
+        private readonly FileStorageContext fileStorageContext;
+
+        public CliRepository(FileStorageContext fileStorageContext)
         {
-            collection = dictionary ?? throw new ArgumentNullException(nameof(dictionary));
+            this.fileStorageContext = fileStorageContext;
         }
         public T GetItem(string itemName)
         {
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/CliRepository.cs]: Done, thx

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.UI.Console/ContainerConfig.cs [hunk 5]
```diff
@@ -26,6 +26,7 @@ public static IContainer Configure()
             builder.RegisterType<CommandFactory>().As<ICommandFactory>();
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.UI.Console/ContainerConfig.cs]: done ,thx

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/Migrations/Configuration.cs (hunk 2 above)
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/Migrations/Configuration.cs]: done, thx

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/SqlConfiguration.cs (hunk 3 above)
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/SqlConfiguration.cs]: Investigated a bit, removed connection factory( thx for tip) still using sqlAzure strategy

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/FileStorageContext.cs (hunk 1 above)
Comment (HannaKeryna) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/FileStorageContext.cs]: No need

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/FileStorageContext.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        public DbSet<UserRepresent> Users { get; set; }
+
+        public FileStorageContext()
+            : base("DBConnection")
+        {
+            Database.SetInitializer(new FileStorageInitializer());
+        }
+        protected override void OnModelCreating(DbModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<FileRepresent>().HasKey(t => t.FileId);
```
Comment (HannaKeryna) [inline Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/FileStorageContext.cs]: I propouse to use separate classes for FileRepresent and UserRepresent models

Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage.Data/FileStorageContext.cs]: fixed in next branch

**Pull request merged**